### DISCUSSION TITLE
Expose event to getEdgeProps

### DISCUSSION THREAD

**Discussion by idl3** [flareact/flareact#82] [General]
**Idea:**
The basic idea is to expose the edge event prop to getEdgeProps. 
This allows us to make application decisions based on the request itself. 

```js
// Flareact: src/workers/index.js:109
const props = await getPageProps(page, query, event);
```

```js
// src/worker/pages.js:106
export async function getPageProps(page, query, event) {
  //...
  if (fetcher) {
    const { props, revalidate } = await fetcher({ params, query: queryObject, event });
  //...
}
```

**Example use case:**
When a user accesses a domain, we want to do a `KV` look-up and append information to an API server request so we can fetch tenanted data.

```js
// simplified KV object representation.
{
  "example.com": "tenant1",
  "cloudflare.com": "tenant2",
  "tenant1-brand1.sharedapp.com": "tenant1",
  "tenant1-brand2.sharedapp.com": "tenant1",
  "tenant2-brand1.sharedapp.com": "tenant2",
  "tenant2-brand2.sharedapp.com": "tenant2"
}
```

```js
// Flareact app: pages/index.js
export async function getEdgeProps({ event }) {
  const requestUrl = event.request.url
  const hostname = new URL(requestUrl).hostname
  const tenantId = await TENANT_HOSTNAMES.get(hostname);

  const responseData = await fetch("API_URL", {
    headers: {
      'X-Tenant': tenantId,
      'Content-Type': 'application/json'
    }
  }).then(response => response.json())

  return {
    props: {
      responseData,
    },
  };
}
```

Let me know what you think?

**Edited based on the discussion below**

**Comment by idl3** [comment]
Ahh, I just realised the flaw in the way I was approaching this. It seems the client-side rendering actually replaces the edge/server-side rendered props. 

I guess I need to dig deeper into the internals and work out a POC first haha.

**Comment by idl3** [comment]
```js
// Flareact app: pages/_app.js
export default function MyApp({ Component, pageProps, edgeEvent }) {
  const requestUrl = edgeEvent ? edgeEvent.request.url : window.location.origin

  const hostname = new URL(requestUrl).hostname
  const tenantId = KV_STORE[hostname];

  return <Component {...pageProps} tenantId={tenantId} />;
}
```

For the specific example, this seems to work as intended.

**Reply by traffisco** [reply]
What's the difference between this and getEdgeProps?
After the initial page load, the html is cached and served to the client

**Reply by idl3** [reply]
There's no real difference actually, I guess to keep concerns consistent we should allow for the edge event to be accessible in getEdgeProps instead.

Let me update the above initial post to reflect the latest implementation.

**Comment by idl3** [comment]
Also just realised that #61 already implements.
